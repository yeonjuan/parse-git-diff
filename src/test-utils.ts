import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
