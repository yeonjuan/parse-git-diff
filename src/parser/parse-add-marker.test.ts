import { createContext } from '../test-utils';
import parseAddMarker from './parse-add-marker';

describe('parseAddMarker', () => {
  it('should parse add marker.', () => {
    const src = `+++ b/src/tests/addition.test.ts`;
    const result = parseAddMarker(createContext(src));

    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });

  it('should not parse delete marker.', () => {
    const src = `--- a/src/tests/addition.test.ts`;
    const result = parseAddMarker(createContext(src));

    expect(result).toBe(null);
  });
});
