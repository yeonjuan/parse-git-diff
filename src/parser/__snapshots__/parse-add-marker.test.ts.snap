// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseAddMarker should parse add marker. 1`] = `"src/tests/addition.test.ts"`;
